From c7272abd136f2764dda161d0f5ff6acd29667c44 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Thu, 3 Mar 2022 16:56:12 +0100
Subject: [PATCH 12/13] sharing/systemd-service: Treat 'static' state as
 enabled

Quoting the documentation:

> static indicates that the unit is statically enabled, i.e. always
> enabled and doesn't need to be enabled explicitly
---
 panels/sharing/cc-systemd-service.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/panels/sharing/cc-systemd-service.c b/panels/sharing/cc-systemd-service.c
index 0b720074e..d2b634676 100644
--- a/panels/sharing/cc-systemd-service.c
+++ b/panels/sharing/cc-systemd-service.c
@@ -112,10 +112,11 @@ cc_is_service_active (const char  *service,
   g_variant_get_child (unit_state_prop, 0, "v", &unit_state_variant);
   unit_state = g_variant_get_string (unit_state_variant, NULL);
 
-  if (g_strcmp0 (unit_state, "enabled") != 0)
+  if (g_strcmp0 (unit_state, "enabled") == 0 ||
+      g_strcmp0 (unit_state, "static") == 0)
+    return TRUE;
+  else
     return FALSE;
-
-  return TRUE;
 }
 
 gboolean
-- 
2.34.1

